pallets/protocol-fee/src/lib.rs

```diff
@@ -208,8 +211,15 @@ impl<T: Config> Module<T> {
     /// Withdraws a precomputed fee from the current payer if it is defined or from the current
     /// identity otherwise.
     fn withdraw_fee(account: T::AccountId, fee: BalanceOf<T>) -> WithdrawFeeResult<T> {
+        // Check if the `account` is being subsidised.
+        let subsidiser = T::Subsidiser::debit_subsidy(&account, fee)
+            .map_err(|_| Error::<T>::InsufficientAccountBalance)?;
```

Comment:
Might be good from an UX perspective to have a distinct error variant for subsidizer failures?